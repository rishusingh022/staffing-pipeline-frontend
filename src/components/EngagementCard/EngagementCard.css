.engagement-card-style img{
    height: 200px !important;
    object-fit: fill;
}