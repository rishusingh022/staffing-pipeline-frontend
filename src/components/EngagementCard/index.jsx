import React from 'react';
import Image from '../Image';
import './EngagementCard.css';
import { BsArrowRight } from 'react-icons/bs';
import PropTypes from 'prop-types';

export default function EngagementCard({
  imageUrl,
  altText,
  identityNumber,
  name,
  startDate,
  status,
  handleButtonClick,
}) {
  return (
    <div className="shadow-lg flex flex-col w-56 engagement-card-style" data-testid="image-card">
      <Image imageUrl={imageUrl} altText={altText} />
      <div className="px-3 py-2">
        <div className="text-xs mb-1">{identityNumber}</div>
        <div className="font-bold text-xl mb-1">{name}</div>
        <div className="font-semibold text-xs mb-1">Start Date: {startDate}</div>
        <div className="font-medium text-xs mb-1 text-gray-500">{status}</div>
      </div>
      <div className="link-button self-end text-xs mb-2 font-medium cursor-pointer" onClick={handleButtonClick}>
        Read More
        <BsArrowRight className="inline-block mx-2 text-blue-800" />
      </div>
    </div>
  );
}

EngagementCard.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
  identityNumber: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};
