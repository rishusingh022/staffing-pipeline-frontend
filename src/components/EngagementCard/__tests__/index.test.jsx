import React from 'react';
// eslint-disable-next-line no-unused-vars
import { fireEvent, screen, render } from '@testing-library/react';
import EngagementCard from '../index';
import { Router } from 'react-router-dom';

jest.mock('react-router-dom');

describe('EngagementCard', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <EngagementCard
          altText={'Image'}
          buttonText={'Know More'}
          handleButtonClick={() => {}}
          identityNumber={'FMNO: 123456'}
          imageUrl={
            'https://images.unsplash.com/photo-1616486497199-1c1e1f1b1b1a?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60'
          }
          name={'John Doe'}
          startDate={'2021-03-01'}
          status={'Active'}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with no image', () => {
    const { asFragment } = render(
      <Router>
        <EngagementCard
          altText={'Image'}
          buttonText={'Know More'}
          handleButtonClick={() => {}}
          identityNumber={'FMNO: 123456'}
          imageUrl={null}
          name={'John Doe'}
          startDate={'2021-03-01'}
          status={'Active'}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // it('should navigate to engagement page on click of read more button', () => {
  //   const navigate = jest.fn();
  //   render(
  //     <Router>
  //       <EngagementCard
  //         altText={'Image'}
  //         buttonText={'Know More'}
  //         handleButtonClick={() => {}}
  //         identityNumber={'FMNO: 123456'}
  //         imageUrl={
  //           'https://images.unsplash.com/photo-1616486497199-1c1e1f1b1b1a?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60'
  //         }
  //         name={'John Doe'}
  //         startDate={'2021-03-01'}
  //         status={'Active'}
  //       />
  //     </Router>

  //   );

  //   const button = screen.getByTestId('read-more-button');
  //   fireEvent.click(button);
  //   expect(navigate).toHaveBeenCalledWith('/projects/FMNO: 123456');
  // });
});
