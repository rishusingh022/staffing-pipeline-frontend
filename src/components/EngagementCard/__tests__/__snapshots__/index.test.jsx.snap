// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngagementCard should render correctly 1`] = `<DocumentFragment />`;

exports[`EngagementCard should render correctly with no image 1`] = `<DocumentFragment />`;
